packages/plugin-table/src/TableRowDomParser.ts

```diff
@@ -0,0 +1,74 @@
+import { AbstractParser } from '../../core/src/AbstractParser';
+import { DomParsingEngine } from '../../plugin-dom/src/DomParsingEngine';
+import { TableRowNode } from './TableRowNode';
+
+export class TableRowDomParser extends AbstractParser<Node> {
+    static id = 'dom';
+    engine: DomParsingEngine;
+
+    predicate = (item: Node): boolean => {
+        return (
+            item instanceof Element &&
+            (item.nodeName === 'THEAD' || item.nodeName === 'TBODY' || item.nodeName === 'TR')
+        );
+    };
+
+    /**
+     * Parse a row node or a table section node.
+     *
+     * @param item
+     */
+    async parse(item: Element): Promise<TableRowNode[]> {
+        if (this._isTableSection(item)) {
+            return this.parseTableSection(item);
+        } else if (item.nodeName === 'TR') {
```

Comment:
It's funny because you're re-playing the predicate but separately this time... ^^'

Reply:
What I meant is that it could be split into two different parsers, but honestly I'm not sure I would like that. Just something I noticed.